libtransact/src/sawtooth.rs

```diff
@@ -462,12 +462,15 @@ mod xo_compat_test {
     }
 
     fn run_schedule(executor: &Arc<Mutex<Option<Executor>>>, scheduler: &mut dyn Scheduler) {
+        let task_iterator = scheduler
+            .take_task_iterator()
+            .expect("Failed to take task iterator");
```

Comment:
Looks like you moved where the panic is, though.

Reply:
What do you mean?